archiver/integrity.py

```diff
@@ -41,6 +51,47 @@ def check_integrity(source_path, deep_flag=False, threads=None, work_dir=None):
     return check_result
 
 
+def check_archive_part_integrity(source_name: Path) -> bool:
+
+    check_result = True
+    for s in [HASH_SUFFIX, TAR_HASH_SUFFIX, LISTING_SUFFIX]:
+        path = source_name.parent / Path(source_name.name + s)
+        if not path.is_file():
+            logging.warning(f"Expected file {path.as_posix()} does not exists.")
+            check_result = False
+    path_c = source_name.parent / Path(source_name.name + COMPRESSED_ARCHIVE_SUFFIX)
+    path_e = source_name.parent / Path(source_name.name + ENCRYPTED_ARCHIVE_SUFFIX)
+    if not (path_c.is_file() or path_e.is_file()):
+        logging.warning(f"Neither of the expected files {path_c.as_posix()} and {path_e.as_posix()} exist.")
+        check_result = False
+
+    path_c = source_name.parent / Path(source_name.name + COMPRESSED_ARCHIVE_HASH_SUFFIX)
+    path_e = source_name.parent / Path(source_name.name + ENCRYPTED_ARCHIVE_HASH_SUFFIX)
+    if not (path_c.is_file() or path_e.is_file()):
+        logging.warning(f"Neither of the expected files {path_c.as_posix()} and {path_e.as_posix()} exist.")
+        check_result = False
+
+    return check_result
+
+def check_archive_list_integrity(source_path: Path) -> bool:
+
+    parts = helpers.get_parts(source_path)
+    source_name = helpers.infer_source_name(source_path)
+
+    logging.info(f'Found {parts} parts in archive {source_path.as_posix()}')
+    check_result = True
+    if parts > 0:
+        for part in range(1, parts + 1):
+            check_result = check_result and check_archive_part_integrity(source_name.parent / Path(source_name.name + f'.part{part}'))
+    else:
+        if len([_ for _ in source_path.glob('*.part*')]) > 0:
```

Comment:
len(list(source_path.glob('*.part*'))) should do it.
In another context, if the generator is very large, you could use `sum(1 for _ in gen)`, but that's a bit harder to read

Reply:
Thanks for the suggestion. I have implemented it this way now.
```
len(list(source_path.glob('*.part*')))
```